### PR TITLE
feat: Remove `getPenalties` getter

### PR DESCRIPTION
**Pull request by deluca-mike**

### CODE REVIEW

File: src/MinterGateway.sol [hunk 1]
```diff
@@ -602,10 +602,70 @@ contract MinterGateway is IMinterGateway, ContinuousIndexing, ERC712Extended {
...
+            //
+            // NOTE: `block.timestamp - missedUntil_` will never be larger than `updateCollateralInterval_` since this
+            //       is only computed after `_getMissedCollateralUpdateParameters`, which ensures that the
+            //       `missedUntil_` is within one `updateCollateralInterval_` of the `block.timestamp`.
+            //
+            // NOTE: `updateCollateralInterval()` never equals 0, so the division is safe.
+            //       Its minimum is capped at `MIN_UPDATE_COLLATERAL_INTERVAL`.
+            uint112 undercollateralizedPenaltyPrincipal_ = UIntMath.bound112(
+                (penaltyRate_ *
+                    ((principalOfExcessOwedM_ * (uint40(block.timestamp) - missedUntil_)) /
```
Comment (toninorair) [inline src/MinterGateway.sol:663]: `block.timestamp` is not accurate, we use min validator timestamp for the selected most optimal solution, we can ask the user to pass min validator timestamp

File: src/MinterGateway.sol [hunk 2]
```diff
@@ -602,10 +602,70 @@ contract MinterGateway is IMinterGateway, ContinuousIndexing, ERC712Extended {
...
+    function getPenalties(
+        address minter_
+    ) external view returns (uint240 missedIntervalPenalty_, uint240 undercollateralizedPenalty_) {
+        uint112 principalOfActiveOwedM_ = principalOfActiveOwedMOf(minter_);
+
+        if (principalOfActiveOwedM_ == 0) return (0, 0);
+
+        uint32 penaltyRate_ = penaltyRate();
+
+        if (penaltyRate_ == 0) return (0, 0);
```
Comment (PierrickGT) [inline src/MinterGateway.sol:614]: We could combine the conditions.
```suggestion
        uint112 principalOfActiveOwedM_ = principalOfActiveOwedMOf(minter_);
        uint32 penaltyRate_ = penaltyRate();
        
        if (principalOfActiveOwedM_ == 0 || penaltyRate_ == 0) return (0, 0);
```

File: src/MinterGateway.sol [hunk 3]
```diff
@@ -602,10 +602,70 @@ contract MinterGateway is IMinterGateway, ContinuousIndexing, ERC712Extended {
...
+
+        MinterState storage minterState_ = _minterStates[minter_];
+
+        (uint40 missedIntervals_, uint40 missedUntil_) = _getMissedCollateralUpdateParameters(
```
Comment (PierrickGT) [inline src/MinterGateway.sol:618]: If `missedIntervals_` is `0`, we should not proceed with the calculation and assign `0` to  `missedIntervalPenalty_` and proceed with the undercollateralization penalty.

File: src/MinterGateway.sol [hunk 4]
```diff
@@ -602,10 +602,70 @@ contract MinterGateway is IMinterGateway, ContinuousIndexing, ERC712Extended {
...
+            uint112 missedIntervalPenaltyPrincipal_ = UIntMath.bound112(
+                (uint256(principalOfActiveOwedM_) * missedIntervals_ * penaltyRate_) / ONE
+            );
+
+            principalOfActiveOwedM_ = UIntMath.bound112(
+                uint256(principalOfActiveOwedM_) + uint256(missedIntervalPenaltyPrincipal_)
+            );
+
+            missedIntervalPenalty_ = _getPresentAmount(missedIntervalPenaltyPrincipal_);
+        }
```
Comment (PierrickGT) [inline src/MinterGateway.sol:635]: A lot of this logic is part of `_imposePenaltyIfMissedCollateralUpdates` and `_imposePenalty`, we should try to create helper functions that would handle the calculations. Same goes for the calculation of `undercollateralizedPenalty_`.

**Review by toninorair — DISMISSED** [review]

**Review by toninorair — APPROVED** [review]

File: src/MinterGateway.sol (hunk 2 above)
Comment (deluca-mike) [reply src/MinterGateway.sol]: Less gas efficient that returning early without an additional couple of SLOADs and calls if `principalOfActiveOwedM_ == 0`.

**Pull request merged**